cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -409,10 +409,13 @@ class StdBasicOStream extends TemplateClass {
 
 /**
  * The `std::ostream` functions `operator<<` (defined as a member function),
- * `put` and `write`.
+ * `put` and `write` and `std::istream::putback`.
  */
 class StdOStreamOut extends DataFlowFunction, TaintFunction {
```

Comment:
It's a bit weird that an `std::istream` member function is modeled by a class named `StdOstreamOut`, but I see why you're doing it.
Is there any downsides to to pull `std::istream::putback` out of this class and create another class with a more suitable name (which could then also include `std::istream::unget` which has a similar functionality)

Reply:
Yeah I think I'll pull it out (the reason I did it this way was that we're starting to get nervous about the volume of code as the models libraries grow and I'm trying to get into the habit of being a bit more efficient, for example using the `[, ]` syntax heavily; in this case I think I took it too far).

Reply:
Done.

Reply:
[`std::istream::unget`](basic_istream& unget();) doesn't need a similar taint model because it's actually a parameterless operation on the stream.  That said, we could possibly do with a model of that and a bunch of similar functions just for the fact they return the qualifier (so you can chain them).  I'll do this in my next PR rather than this one.

Reply:
Addressed `unget` in the fourth commit of https://github.com/github/codeql/pull/4283.